knime_extension/src/nodes/spatialnetwork.py

```diff
@@ -125,22 +125,22 @@ def fetch_google_OD(download_link):
                 print("Empty value generated")
             return [nt_time, nt_dist]
 
-        O_gdf = left_input.to_pandas().rename(columns={self.O_geo_col: "geometry"})
-        D_gdf = right_input.to_pandas().rename(columns={self.D_geo_col: "geometry"})
-        O_gdf = gp.GeoDataFrame(O_gdf, geometry="geometry")
-        D_gdf = gp.GeoDataFrame(D_gdf, geometry="geometry")
+        O_gdf = knut.load_geo_data_frame(left_input, self.O_geo_col, exec_context)
+        D_gdf = knut.load_geo_data_frame(right_input, self.D_geo_col, exec_context)
+        O_gdf = O_gdf[[self.O_geo_col]].rename(columns={self.O_geo_col: "geometry"})
```

Comment:
what happens if a column with name geometry already exists? Why do you need to rename the column? Can't you use the O_geo_col and D_geo_col variable instead?

Reply:
as stated above,  new data was created to avoid such scenarios